File: 투포인터/수들의합5.java
package 투포인터;

import java.util.Scanner;

public class 수들의합5 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        int startNum = 1;
        int lastNum = 1;
        int count = 0;
        int currentSum = 1;

        while (true) {
            if (currentSum < n && lastNum >= n + 1) {
                break;
            }

            if (currentSum < n) {
                lastNum++;
                currentSum += lastNum;
            } else if (currentSum > n) {
                currentSum -= startNum;
                startNum++;
            } else if (currentSum == n) {
                count++;
                currentSum -= startNum;
                startNum++;
            }
        }

        System.out.println(count);
    }
}

File: 투포인터/수들의합5.py
n = int(input())

start_num = 1
last_num = 1
count = 0
current_sum = 1


while(True):
    if (current_sum < n and last_num >= n + 1 ):
        break

    if current_sum < n:
        last_num += 1
        current_sum += last_num
    elif current_sum > n:
        current_sum -= start_num
        start_num += 1
    elif current_sum == n:
        count += 1
        current_sum -= start_num
        start_num += 1

print(count)